### ISSUE TITLE
MTE crash when syncing monero wallet

### ISSUE THREAD

**Issue by gsdgethjfjj** [cake-tech/cake_wallet#1515]
**Describe the bug**
App crashes when I'm trying to sync monero wallet with memory tagging enabled. 

**To Reproduce**
Open monero wallet and try to sync

**Expected behavior**
No crashes

**Logs**
```
type: crash
osVersion: google/husky/husky:14/AP2A.240705.005/2024070201:user/release-keys
uid: 10236 (u:r:untrusted_app_32:s0:c236,c256,c512,c768)
cmdline: com.cakewallet.cake_wallet
processUptime: 183s

signal: 11 (SIGSEGV), code 9 (SEGV_MTESERR), faultAddr 900c5a6318f6800
threadName: DartWorker
MTE: enabled

backtrace:
    /apex/com.android.runtime/lib64/bionic/libc.so (__memcpy_aarch64_simd+36, pc 5f4e4)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk!libcw_monero.so (pc 85fe00)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk!libcw_monero.so (std::__ndk1::basic_ostream<char, std::__ndk1::char_traits<char> >::write(char const*, long)+176, pc 8ed3b0)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk!libcw_monero.so (bool do_serialize_container<binary_archive, std::__ndk1::deque<crypto::hash, std::__ndk1::allocator<crypto::hash> > >(binary_archive<true>&, std::__ndk1::deque<crypto::hash, std::__ndk1::allocator<crypto::hash> >&)+364, pc aa7fec)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk!libcw_monero.so (bool tools::hashchain::do_serialize_object<true, binary_archive>(binary_archive<true>&)+348, pc aa7e44)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk!libcw_monero.so (bool tools::wallet2::do_serialize_object<true, binary_archive>(binary_archive<true>&)+284, pc aa7768)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk!libcw_monero.so (tools::wallet2::get_cache_file_data(epee::wipeable_string const&)+296, pc 957448)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk!libcw_monero.so (tools::wallet2::store_to(std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char> > const&, epee::wipeable_string const&)+932, pc 9555d8)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk!libcw_monero.so (tools::wallet2::store()+80, pc 9368d4)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk!libcw_monero.so (Monero::WalletImpl::store(std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char> > const&)+176, pc 846820)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk!libcw_monero.so (store+212, pc 83b9a0)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk (pc 801598)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk (pc b76ab4)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk (pc e47870)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk (pc e477fc)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk (pc 974604)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk (pc 973fac)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk (pc 973f3c)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk (pc 1337844)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk (pc 810140)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk (pc 1337844)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk (pc 80f26c)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk (pc 803d2c)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk!libflutter.so (pc 83b988)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk!libflutter.so (pc 847318)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk!libflutter.so (pc 84c548)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk!libflutter.so (pc 84c30c)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk!libflutter.so (pc 8e1254)
    /data/app/~~JguAKDEE5h6iU0pfhxjHtA==/com.cakewallet.cake_wallet-w_-QX-83M6ZCNgaeCUUh2Q==/base.apk!libflutter.so (pc 8a6900)
    /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+204, pc 795dc)
    /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+68, pc 69fa4)
```

**Platform:**
Graphene OS, Pixel 8 Pro
Cake Wallet Version: 4.18.2

**Additional context**
Not sure if it's not a duplicate of https://github.com/cake-tech/cake_wallet/issues/1199 but still.

**Comment by tuxpizza** [comment]
Hi, thanks for the report. This is a bug with the existing Monero library in that version. Running into this error should be greatly reduced in the current pre-release (4.19.0) which is available on Github and will be rolled out to Play Store and App Store users over the next week.

Feel free to try the pre-release, but if you do I'd recommend creating a backup, just incase something goes wrong as the 4.19.0 has major changes with the Monero libraries.

**Comment by tuxpizza** [comment]
Fixed in 4.19.0
(Feel free to open this back up if you run into the same error again)
